### DISCUSSION TITLE
Fetching data on direct link entrance

### DISCUSSION THREAD

**Discussion by irakliShengelia** [nuxt/nuxt#10168] [Questions (Nuxt 2)]
What is the best way of fetching data on direct link entrance?

I'm using nuxt 3, asyncData hook - it works fine when I navigate to the page via nuxt-link, however, when I copy the link and open it in a new tab - the asyncData hook doesn't execute at all.

How do you usually address this issue? Is this a bug? It doesn't make sense for a universal application to not fetch data on direct link navigation - I'm probably missing some info.

Env:
Nuxt 3
SSR
Nuxt Config:
`export default {
  // Global page headers: https://go.nuxtjs.dev/config-head
  head: {
    title: 'tourfront',
    htmlAttrs: {
      lang: 'en',
    },
    meta: [
      { charset: 'utf-8' },
      { name: 'viewport', content: 'width=device-width, initial-scale=1' },
      { hid: 'description', name: 'description', content: '' },
      { name: 'format-detection', content: 'telephone=no' },
    ],
    link: [{ rel: 'icon', type: 'image/x-icon', href: '/favicon.ico' }],
  },

  // Global CSS: https://go.nuxtjs.dev/config-css
  css: [],

  // Plugins to run before rendering page: https://go.nuxtjs.dev/config-plugins
  plugins: [
    { src: '~/plugins/datepicker.js', mode: 'client' }
  ],

  // Auto import components: https://go.nuxtjs.dev/config-components
  components: true,

  // Modules for dev and build (recommended): https://go.nuxtjs.dev/config-modules
  buildModules: [
    // https://go.nuxtjs.dev/typescript
    '@nuxt/typescript-build',
    // https://go.nuxtjs.dev/tailwindcss
    '@nuxtjs/tailwindcss',
    '@nuxtjs/moment',
  ],

  // Modules: https://go.nuxtjs.dev/config-modules
  modules: [
    // https://go.nuxtjs.dev/axios
    '@nuxtjs/axios',
  ],

  // Axios module configuration: https://go.nuxtjs.dev/config-axios
  axios: {
    // Workaround to avoid enforcing hard-coded localhost:3000: https://github.com/nuxt-community/axios-module/issues/308
    baseURL: process.env.BASE_URL,
  },

  // Build Configuration: https://go.nuxtjs.dev/config-build
  build: {},
}
`

**Comment by DennisAugusto** [comment]
Hi @irakliShengelia! I got the same issue, have you solved?
